homework/index.js

```diff
@@ -27,21 +12,151 @@
       }
     });
+
     return elem;
   }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  async function fetchJSON(url) {
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`Error!!! : ${response.status}-${response.statusText}`);
       }
+      const parsedResponse = await response.json();
+      return parsedResponse;
+    } catch (error) {
+      const root = document.getElementById('root');
+      createAndAppend('div', root, {
+        text: error.message,
+        class: 'alert-error',
+      });
+      return false;
+    }
+  }
+
+  function addRow(parent, labelText, rowValue) {
+    const row = createAndAppend('tr', parent);
+    createAndAppend('td', row, {
+      class: 'label',
+      text: labelText,
+    });
+    createAndAppend('td', row, {
+      text: rowValue,
+    });
+    return row;
+  }
+
+  function renderRepositoryInfo(repository, leftContainer) {
+    const table = createAndAppend('table', leftContainer);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository : ', '');
+    createAndAppend('a', firstRow.children[1], {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    addRow(tbody, 'Description :', repository.description);
+    addRow(tbody, 'Forks :', repository.forks);
+    addRow(tbody, 'Updated :', repository.updated_at);
+  }
+
+  function renderRepoContributors(contributors, rightContainer) {
+    createAndAppend('p', rightContainer, {
+      class: 'contributor-header',
+      text: 'Contributors',
+    });
+    const ul = createAndAppend('ul', rightContainer, {
+      class: 'contributor-list',
+    });
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-container',
+      });
+      createAndAppend('img', li, {
+        class: 'contributor-avatar',
+        src: contributor.avatar_url,
+      });
+      const contributorDataDiv = createAndAppend('div', li, {
+        class: 'contributor-data',
+      });
+      createAndAppend('a', contributorDataDiv, {
+        text: contributor.login,
+        href: contributor.html_url,
+        class: 'contributor-name',
+      });
+      createAndAppend('div', contributorDataDiv, {
+        text: contributor.contributions,
+        class: 'contributionCount',
+      });
     });
   }
 
+  async function fetchContributors(url, rightContainer) {
+    try {
+      const fetchedContributors = await fetchJSON(url);
+      renderRepoContributors(fetchedContributors, rightContainer);
+    } catch (error) {
+      createAndAppend('div', rightContainer, {
+        text: `We got an error: $ {error.message}`,
```

Comment:
Oops! 
This won't work 😅
```suggestion
        text: `We got an error: ${error.message}`,
```